### PR TITLE
Db refactoring

### PR DESCRIPTION
**Pull request by MarekM25**

### CODE REVIEW

File: src/Nethermind/Nethermind.Baseline/BaselineModuleFactory.cs [hunk 1]
```diff
@@ -74,8 +75,8 @@ public override IBaselineModule Create()
                 _blockFinder,
                 _abiEncoder,
                 _fileSystem,
-                _dbProvider.BaselineTreeDb,
-                _dbProvider.BaselineTreeMetadataDb,
+                _dbProvider.GetDb<IDb>(BaselineDbConsts.BaselineTreeDbName),
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Baseline/BaselineModuleFactory.cs:78]: (minor) maybe:

BaselineDbNames.Tree
BaselineDbNames.TreeMetadata

File: src/Nethermind/Nethermind.Baseline/Database/BaselineDbInitializer.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+namespace Nethermind.Baseline.Database
+{
+    public class BaselineDbConsts
+    {
+        public const string BaselineTreeDbName = "baselineTree";
+        public const string BaselineTreeMetadataDbName = "baselineTreeMetadata";
+    }
+
+    public class BaselineDbInitializer : RocksDbInitializer
+    {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Baseline/Database/BaselineDbInitializer.cs:31]: superb

File: src/Nethermind/Nethermind.Baseline/Database/Metrics.cs [hunk 3]
```diff
@@ -14,21 +14,22 @@
...
-        public BlocksRocksDb(string basePath, IDbConfig dbConfig, ILogManager logManager = null)
-            : base(basePath, "blocks", dbConfig, logManager)
-        {
-        }
+        [Description("Number of baseline tree DB writes.")]
+        public static long BaselineTreeDbWrites { get; set; }
 
-        protected internal override void UpdateReadMetrics() => Metrics.BlocksDbReads++;
-        protected internal override void UpdateWriteMetrics() => Metrics.BlocksDbWrites++;
+        [Description("Number of baseline tree (faucet) DB reads.")]
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Baseline/Database/Metrics.cs:29]: (faucet?)

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Baseline/Database/Metrics.cs:29]: remove faucet - was it taken from NDM?

File: src/Nethermind/Nethermind.DataMarketplace.Consumers.Infrastructure/NdmConsumersModule.cs [hunk 4]
```diff
@@ -150,15 +150,17 @@ public void Init()
...
                 default:
-                    depositRepository = new DepositDetailsRocksRepository(rocksDbProvider.DepositsDb,
+                    var dbInitializer = new ConsumerNdmDbInitializer(_api.DbProvider, ndmConfig, _api.RocksDbFactory, _api.MemDbFactory);
+                    await dbInitializer.Init();
+                    depositRepository = new DepositDetailsRocksRepository(_api.DbProvider.GetDb<IDb>(ConsumerNdmDbConsts.DepositsDbName),
                         depositDetailsRlpDecoder);
                     depositApprovalRepository = new ConsumerDepositApprovalRocksRepository(
-                        rocksDbProvider.ConsumerDepositApprovalsDb, depositApprovalRlpDecoder);
-                    providerRepository = new ProviderRocksRepository(rocksDbProvider.DepositsDb,
+                        _api.DbProvider.GetDb<IDb>(ConsumerNdmDbConsts.ConsumerDepositApprovalsDbName), depositApprovalRlpDecoder);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.DataMarketplace.Consumers.Infrastructure/NdmConsumersModule.cs:158]: _api.Db(name) => _api.GetDb<IDb>(...)
_api.Db<T>(name) => _api.DbProvider.GetDb<T>(...)
===
similar as for configs maybe?

**Review by tkstanczak — COMMENTED** [review]
this is beautiful, crying of joy

File: src/Nethermind/Nethermind.Baseline/Database/Metrics.cs (hunk 3 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.Baseline/Database/Metrics.cs:29]: > (faucet?)

copy paste error :) thx for catch

**Review by tkstanczak — APPROVED** [review]
worked with this code on another branch, solid, some minor changes leaving for later - it will improve the codebase a lot

**Pull request merged**